lms/djangoapps/course_api/blocks/forms.py

```diff
@@ -0,0 +1,129 @@
+"""
+TODO
+"""
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.forms import Form, CharField, ChoiceField, IntegerField
+from django.http import Http404
+from rest_framework.exceptions import PermissionDenied
+
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import UsageKey
+from openedx.core.djangoapps.util.forms import MultiValueField
+from xmodule.modulestore.django import modulestore
+
+from .permissions import can_access_other_users_blocks, can_access_users_blocks
+
+
+class BlockListGetForm(Form):
+    """
+    A form to validate query parameters in the block list retrieval endpoint
+    """
+    user = CharField(required=True)  # TODO return all blocks if user is not specified by requesting staff user
+    usage_key = CharField(required=True)
+    requested_fields = MultiValueField(required=False)
+    student_view_data = MultiValueField(required=False)
+    block_counts = MultiValueField(required=False)
+    depth = CharField(required=False)
+    nav_depth = IntegerField(required=False, min_value=0)
+    return_type = ChoiceField(
+        required=False,
+        choices=[(choice, choice) for choice in ['dict', 'list']],
+    )
+
+    def clean_requested_fields(self):
+        """
+        TODO
+        """
+        requested_fields = self.cleaned_data['requested_fields']
+
+        # add default requested_fields
+        return (requested_fields or set()) | {'type', 'display_name'}
+
+    def clean_depth(self):
+        """
+        TODO
+        """
+        value = self.cleaned_data['depth']
+        if not value:
+            return 0
+        elif value == "all":
+            return None
+        try:
+            return int(value)
+        except ValueError:
+            raise ValidationError("'{}' is not a valid depth value.".format(value))
+
+    def clean_usage_key(self):
+        """
+        TODO
+        """
+        usage_key = self.cleaned_data['usage_key']
+
+        try:
+            usage_key = UsageKey.from_string(usage_key)
+            usage_key = usage_key.replace(course_key=modulestore().fill_in_run(usage_key.course_key))
+        except InvalidKeyError:
+            raise ValidationError("'{}' is not a valid usage key.".format(unicode(usage_key)))
+
+        return usage_key
+
+    def clean_return_type(self):
+        """
+        Return valid 'return_type' or default value of 'dict'
+        """
+        return self.cleaned_data['return_type'] or 'dict'
+
+    def clean_requested_user(self, cleaned_data, course_key):
+        """
+        Validates and returns the requested_user, while checking permissions.
+        """
+        requested_username = cleaned_data.get('user', '')
```

Comment:
Probably not necessary, but given the TODO note on the user field above, I'm inclined to leave it in place to make later modifications easier.